picard/util/__init__.py

```diff
@@ -380,3 +381,32 @@ def album_artist_from_path(filename, album, artist):
             elif not artist and len(dirs) >= 2:
                 artist = dirs[-2]
     return album, artist
+
+
+def build_qurl(host, port=80, path=None, mblogin=False, queryargs=None):
+    """
+    Builds and returns a QUrl object from `host`, `port` and `path` and
+    automatically enables HTTPS if necessary.
+
+    Setting `mblogin` to True forces HTTPS.
+
+    Encoded query arguments can be provided in `queryargs`, a
+    dictionary mapping field names to values.
+    """
+    url = QtCore.QUrl()
+    url.setHost(host)
+    url.setPort(port)
+    if (mblogin or  # Login is required
+        port == 443 or  # HTTPS port
+        (host in MUSICBRAINZ_SERVERS and port == 80)):  # We know MB supports HTTPS
+            url.setScheme("https")
+            url.setPort(443)
```

Comment:
Using this adds `:443` to the URL even though HTTPS implies this.

My personal preference would be to add code to only set the port when it is not default (see the code in my previous fix) - but it is equally valid to keep the code simple and have the port added each time.